diff --git a/lib/Test/Vitest.js b/lib/Test/Vitest.js
index 0ecf9ccff01bb6556bf3e471d5a93254d16224a7..c0f1237e6a98e698718b0bbde5de8eb58c859e8e 100644
--- a/lib/Test/Vitest.js
+++ b/lib/Test/Vitest.js
@@ -3,7 +3,7 @@ import { it } from "@effect/vitest";
 import * as Effect from "effect/Effect";
 import * as Exit from "effect/Exit";
 import * as Scope from "effect/Scope";
-import { afterAll as vitestAfterAll, afterEach as vitestAfterEach, beforeAll as vitestBeforeAll, beforeEach as vitestBeforeEach, } from "vitest";
+import { afterAll as vitestAfterAll, afterEach as vitestAfterEach, beforeAll as vitestBeforeAll, beforeEach as vitestBeforeEach, } from "vite-plus/test";
 import * as Core from "./Core.js";
 export { executeWhenReady, getWhenReady, guardContentType, guardedFetchLayer, rpcClientLayer, WorkerNotReady, } from "./Http.js";
 export const defaultStage = Core.defaultStage;
diff --git a/src/Test/Vitest.ts b/src/Test/Vitest.ts
index c321e6d4924d0f6b3ae53292a12af88167692ccb..673929a81d066bdd6a2dd94e9c1d95bb69f3e3cd 100644
--- a/src/Test/Vitest.ts
+++ b/src/Test/Vitest.ts
@@ -8,7 +8,7 @@ import {
   afterEach as vitestAfterEach,
   beforeAll as vitestBeforeAll,
   beforeEach as vitestBeforeEach,
-} from "vitest";
+} from "vite-plus/test";
 
 import type { AlchemyContext } from "../AlchemyContext.ts";
 import type { CompiledStack } from "../Stack.ts";
