diff --git a/ios/Core/SharedObjects/SharedObjectRegistry.swift b/ios/Core/SharedObjects/SharedObjectRegistry.swift
index bb2bdcdc3b93ae9dea7dbc4c4f7e36109590d5c4..acafea7564cfec9a4b5bb8c37d6b09e13484b2af 100644
--- a/ios/Core/SharedObjects/SharedObjectRegistry.swift
+++ b/ios/Core/SharedObjects/SharedObjectRegistry.swift
@@ -34,7 +34,7 @@ public final class SharedObjectRegistry: Sendable {
   /**
    Weak reference to the app context for the registry.
    */
-  private weak let appContext: AppContext?
+  private nonisolated(unsafe) weak var appContext: AppContext?
 
   internal struct State: Sendable {
     /**
